doc/kubernetes/modules/ROOT/pages/automation-using-tool-task.adoc

```diff
@@ -0,0 +1,97 @@
+= Automation using the tool _task_
+:description: Few details around the Automation related to provisioning and configuration of the Kubernetes deployment.
+
+{description}
+
+== Why _this_ Automation tool
+
+There are several dependencies that need to be built and executed in the right order, for this Kubernetes provisioning to work properly.
+To simplify upgrades and fast iterations during development, all steps and dependencies have been scripted using https://taskfile.dev/installation/#get-the-binary[_task_].
+Think of this tool as a modern version of _make_ that allows simple check-summing of input files and parallel execution of tasks.
+
+== How does the _task_ tool work
+
+All tasks are described in the `Taskfile.yaml` file.
+If one of the commands in a task fails, the whole task fails.
+If there are parallel tasks running and one of the tasks fails, _task_ kills the other tasks running in parallel.
+
+For environment-specific settings, add a `.env` file with the contents necessary for your environment.
+
+== Now some _task_ commands in action
+
+The following list shows some command line examples:
+
+`task`::
+Executes the `default` task, which updates the minikube installation with the latest changes.
+Run it after every local change to a file, or after pulling changes from upstream via git.
+
+`task -f`::
+Executes the `default` task, but execute all tasks event if no source file has been changed.
+Run it after minikube has been re-created.
+
+`task <taskname>`::
+Execute a specific task from the `Taskfile.yaml`.
+Most tasks are set up to run only when modified, so task might reply with `task: Task "<taskname>" is up to date`.
+To force execution of a task, add the `-f` flag.
+This executes both the task and its dependencies.
+
+`task <var>=<value>`::
+Set a variable with a specific value, then run the task.
+Use it for example to set the storage type in a one-off run: `task KC_STORAGE=jpa`.
+
+`task --dry`::
+Show which tasks would be executed.
+Run it to see what commands _task_ would execute on the next run.
+Can be combined with a task name and the `-f` flag.
+
+`task <taskname> \-- <cli_args>`::
+Execute a specific task from the `Taskfile.yaml` by passing through the command line args that would be needed by the task file. This allows the users to re-use their shell scripts or other programs without having to re-implement them in the specific task.
++
+An Example of such task would be to find out the last completed job of the Dataset provider,
++
```

Comment:
I added `+` signs here as continuations where the sections are part of that item and keep the indentation.

Reply:
makes sense, I will keep a note from next time